### DISCUSSION TITLE
Have a rolling backup of the user's beets db? How big is the average beets database?

### DISCUSSION THREAD

**Discussion by RollingStar** [beetbox/beets#5339] [Ideas]
I don't see any backup related code in the codebase. The user shouldn't rely on us for backups, but on the other hand, how hard would it be to make `library_backup.db` by just copying the db ahead of certain actions? Even easier would be copying each time beets is run, perhaps with a date check so it only copies once per day. With 522 albums I have a 72 MB library. That's [140 KB per album](https://www.wolframalpha.com/input?i=72+MB+%2F+522) with some plugins and stuff.

**Comment by Serene-Arc** [comment]
Since beets uses sqlite, the database is just a file. Copying it copies the entirety of the database at that time. As long as you aren't literally running a beets command at that moment, you can just copy and paste it through any interface and it's a backup. I think forcing backups on the user in weird ways would be a bad thing to do, especially since we don't actually have any way to properly back it up in the technical sense without some user input on the process

A backup in terms of beets code would just be a `shutil.copy` call. Doing this as part of the base program isn't something I'd recommend but maybe a plugin, so it's opt-in? Something that can be configured to copy to a specific place at a specific interval, or be set before or after other commands are run. Like a 'just in case' when you're doing a big `modify` command or something.
